lib/srv/desktop/deskproto/proto.go

```diff
@@ -0,0 +1,254 @@
+// Package deskproto implements the desktop protocol encoder/decoder.
+// See https://github.com/gravitational/teleport/blob/master/rfd/0037-desktop-access-protocol.md
+//
+// TODO(awly): complete the implementation of all messages, even if we don't
+// use them yet.
+package deskproto
+
+import (
+	"bytes"
+	"encoding/binary"
+	"image"
+	"image/png"
+	"io"
+
+	"github.com/gravitational/trace"
+)
+
+// MessageType identifies the type of the message.
+type MessageType byte
+
+const (
+	TypeClientScreenSpec = MessageType(1)
```

Comment:
Might be helpful to add godocs on these to explain what they represent. Some are self-explanatory, some are not.